.golangci.yml

```diff
@@ -711,6 +712,18 @@ linters-settings:
       - name: var-naming
       # non-default rules:
       - name: duplicated-imports
+  forbidigo:
+    forbid:
+      - p: ^model.NewConfig.*$
+        pkg: ^github.com/DataDog/datadog-agent/pkg/config/model$
+        msg: use pkg/config/mock instead in tests or the config component
+      - p: ^setup.SetDatadog.*$
+        pkg: ^github.com/DataDog/datadog-agent/pkg/config/setup$
+        msg: use pkg/config/mock instead for tests
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
This assume calls could come only from tests, is this linter limited to `_test` files? Should you suggest using a component in agent code?

Reply:
Would it make sense to add the build tag `//go:build test` to the mock package? That way we warranty that is not used outside of tests?

https://github.com/DataDog/datadog-agent/blob/fa0359476f6f1f4b2393859a9dcc372958393f43/pkg/config/mock/mock.go#L7

Reply:
That would prevent outside repo like OTEL/Serverless from using the mock without themselves using the `test` build tag. Add build tags is not always possible, especially in open-source repo like OTEL.